pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/periodictask/ControllerPeriodicTask.java

```diff
@@ -33,16 +33,20 @@ public ControllerPeriodicTask(String taskName, long runFrequencyInSeconds,
     this(taskName, runFrequencyInSeconds, DEFAULT_INITIAL_DELAY_IN_SECOND, pinotHelixResourceManager);
   }
 
-  public ControllerPeriodicTask(String taskName, long runFrequencyInSeconds, long initialDelaySeconds,
+  public ControllerPeriodicTask(String taskName, long runFrequencyInSeconds, long initialDelayInSeconds,
       PinotHelixResourceManager pinotHelixResourceManager) {
-    super(taskName, runFrequencyInSeconds, initialDelaySeconds);
+    super(taskName, runFrequencyInSeconds, initialDelayInSeconds);
     _pinotHelixResourceManager = pinotHelixResourceManager;
   }
 
+  @Override
+  public void init() {
+  }
+
   @Override
   public void run() {
     if (!_pinotHelixResourceManager.isLeader()) {
-      onBecomeNotLeader();
+      nonLeaderCleanUp();
```

Comment:
We should call this only once in the lifetime until the controller assumes leadership again. So, the base class needs to remember if it has already invoked the method. 

I would also advocate having a similar method onBecomeLeader() so that a task can implement one call that needs to happen when it gains leadership.